cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -119,7 +119,7 @@ public ScheduleTaskRunner(Store store, ProgramLifecycleService lifecycleService,
    *
    * @return a {@link ListenableFuture} object that completes when the program completes
    */
-  private ListenableFuture<?> execute(final ProgramId id, Map<String, String> sysArgs,
+  public ListenableFuture<?> execute(final ProgramId id, Map<String, String> sysArgs,
```

Comment:
the `run` above checks whether the appSpec contains scheduleSpec and whether scheduleSpec contains the program